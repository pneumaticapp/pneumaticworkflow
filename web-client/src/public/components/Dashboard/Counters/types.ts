/* eslint-disable */
/* prettier-ignore */
export enum EDashboardCounterType {
  Started = 'started',
  InProgress = 'in-progress',
  Overdue = 'overdue',
  Completed = 'completed',
}
