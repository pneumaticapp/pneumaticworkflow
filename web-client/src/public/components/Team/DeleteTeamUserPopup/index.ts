/* eslint-disable */
/* prettier-ignore */
export * from './DeleteTeamUserPopup';
export * from './container';
