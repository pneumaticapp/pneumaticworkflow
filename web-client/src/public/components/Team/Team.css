/* stylelint-disable declaration-no-important */
.container {
  margin: 0 auto;
  max-width: 864px;
}

.search {
  position: relative;
  margin-bottom: 32px;
  width: 100%;
}

.search__icon {
  position: absolute;
  top: 50%;
  margin-left: 16px;
  color: var(--pneumatic-color-black72);
  transform: translateY(-50%);
}

.search-field {
  margin-bottom: 32px;
}

.search-field__input {
  padding-left: 48px !important;
  background-color: var(----background-color);
}

.cards {
  width: 100%;
}

.card-wrapper:not(:last-child) {
  margin-bottom: 8px;
}

.card {
  position: relative;
  padding: 32px 160px 32px 32px;
  display: flex;
  min-height: 104px;
  color: var(--pneumatic-color-black100);
  background-color: white;
  border-radius: 24px;
  align-items: center;
  transition: box-shadow 0.2s ease-in-out;

  &:hover {
    box-shadow: 0 8px 24px rgba(98, 98, 95, 0.08);
  }

  @media (--mobile) {
    flex-direction: column;
    align-items: flex-start;
  }
}

.card-aside {
  position: absolute;
  top: 32px;
  right: 32px;
  display: flex;
  height: 40px;
  align-items: center;
}

.card-admin {
  display: flex;
  align-items: center;
  font-size: 11px;
  font-weight: bold;
  line-height: 16px;
  text-align: right;
  letter-spacing: 0.05em;
  text-transform: uppercase;

  @media (--mobile) {
    top: calc(100% - 48px);
    right: 16px;
  }
}

.card-admin__switch {
  margin-left: 8px;
}

.card-more-container {
  margin-left: 16px;
}

.card-more {
  cursor: pointer;
  color: var(--pneumatic-color-black16);
  transition: color 0.1s ease-in-out;

  &.card-more_active,
  &:hover {
    color: var(--pneumatic-color-link-hover);
  }
}

.card-header {
  position: relative;
  margin: 0;
}

.card-subheader {
  margin: 0;
  font-size: 13px;
  line-height: 16px;
  color: var(--pneumatic-color-black48);
}

.card-avatar {
  margin-right: 16px;

  @media (--mobile) {
    margin-right: 0;
    margin-bottom: 16px;
  }
}

.card-badge {
  margin: 0;
  margin-right: 40px;
  font-size: 11px;
  font-weight: bold;
  line-height: 16px;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  color: var(--pneumatic-color-black32);
}

.card-menu-icon {
  color: var(--pneumatic-color-black72);
}

.invite-button {
  all: unset;
  margin-bottom: 8px;
  display: flex;
  width: 100%;
  height: 104px;
  color: var(--pneumatic-color-link-dark);
  background: white;
  border-radius: 24px;
  justify-content: center;
  align-items: center;
  transition: all 0.2s ease-in-out;

  &:hover {
    color: var(--pneumatic-color-link-hover);
    box-shadow: 0 8px 24px rgba(98, 98, 95, 0.08);
  }

  @media (--mobile) {
    margin-left: 0;
  }
}

.invite-button__icon {
  margin-right: 8px;
}
