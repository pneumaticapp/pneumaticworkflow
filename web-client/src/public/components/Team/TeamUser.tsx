import * as React from 'react';
import * as classnames from 'classnames';
import Switch from 'rc-switch';
import { debounce } from 'throttle-debounce';
import { useIntl } from 'react-intl';

import { EUserStatus, TUserListItem } from '../../types/user';
import { IntlMessages } from '../IntlMessages';
import { Avatar } from '../UI/Avatar';
import { getUserFullName } from '../../utils/users';
import { DateFormat } from '../UI/DateFormat';
import { Dropdown, Header, TDropdownOption } from '../UI';
import { AddUserIcon, MoreIcon, RemoveUserIcon, TrashIcon } from '../icons';

import styles from './Team.css';

export interface ITeamUserProps {
  user: TUserListItem;
  isCurrentUser?: boolean;
  isSubscribed?: boolean;
  resendInvite(): Promise<void>;
  handleToggleAdmin(user: TUserListItem): () => Promise<void>;
  openModal(): void;
}

export function TeamUser(props: ITeamUserProps) {
  const {
    user,
    user: { isAdmin, isAccountOwner, status, email, invite },
    handleToggleAdmin,
    resendInvite,
    isCurrentUser,
    isSubscribed,
    openModal,
  } = props;

  const { formatMessage } = useIntl();
  const isUserActive = status === EUserStatus.Active;

  const resendInviteDebounced = React.useCallback(debounce(700, resendInvite), [resendInvite]);

  const dropdownOptions: TDropdownOption[] = [
    {
      label: formatMessage({ id: 'team.card-resend-invite' }),
      onClick: resendInviteDebounced,
      isHidden: isUserActive,
      Icon: AddUserIcon,
    },
    {
      label: isUserActive
        ? formatMessage({ id: 'team.card-delete' })
        : formatMessage({ id: 'team.card-revoke-invite' }),
      onClick: openModal,
      Icon: isUserActive ? TrashIcon : RemoveUserIcon,
      color: 'red',
      withUpperline: !isUserActive,
    },
  ];

  const renderDetails = () => {
    const detailsMap: { [key in string]: JSX.Element } = {
      [EUserStatus.Active]: (
        <>
          <Header size="6" tag="p" className={styles['card-header']}>
            {getUserFullName(user)}
          </Header>
          <p className={styles['card-subheader']}>
            <strong>{formatMessage({ id: 'team.user-id' }, { id: user.id })}</strong>{' '}
            {formatMessage({ id: 'team.user-email' }, { email: user.email })}
          </p>
        </>
      ),
      [EUserStatus.Invited]: (
        <>
          <Header size="6" tag="p" className={styles['card-header']}>
            {email}
          </Header>
          <p className={styles['card-subheader']}>
            {formatMessage(
              {
                id: 'team.user-invited',
              },
              {
                date: <DateFormat date={invite?.dateCreated} />,
                username: invite?.byUsername,
              },
            )}
          </p>
        </>
      ),
    };

    const details = detailsMap[status];

    if (!details) {
      return null;
    }

    return <div>{details}</div>;
  };

  const renderControllers = () => {
    const controllersMap = [
      {
        check: () => isCurrentUser,
        render: () => <p className={styles['card-badge']}>{formatMessage({ id: 'team.card-current-user-badge' })}</p>,
      },
      {
        check: () => isAccountOwner,
        render: () => <p className={styles['card-badge']}>{formatMessage({ id: 'team.card-account-owner-badge' })}</p>,
      },
      {
        check: () => true,
        render: () => (
          <>
            <div className={styles['card-admin']}>
              <IntlMessages id="team.card-admin-toggle" />
              <Switch
                className={classnames(
                  'custom-switch custom-switch-primary custom-switch-small',
                  styles['card-admin__switch'],
                )}
                onChange={handleToggleAdmin(user)}
                checked={isSubscribed ? isAdmin : true}
                checkedChildren={null}
                unCheckedChildren={null}
              />
            </div>
            <div className={styles['card-more-container']}>
              <Dropdown
                renderToggle={(isOpen) => (
                  <MoreIcon className={classnames(styles['card-more'], isOpen && styles['card-more_active'])} />
                )}
                options={dropdownOptions}
              />
            </div>
          </>
        ),
      },
    ];

    return <div className={styles['card-aside']}>{controllersMap.find(({ check }) => check())?.render()}</div>;
  };

  return (
    <div className={styles['card-wrapper']}>
      <div className={classnames(styles['card'])}>
        <Avatar user={user} containerClassName={styles['card-avatar']} size="lg" />
        {renderDetails()}
        {renderControllers()}
      </div>
    </div>
  );
}
