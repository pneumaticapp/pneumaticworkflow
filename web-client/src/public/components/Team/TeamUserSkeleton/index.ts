/* eslint-disable */
/* prettier-ignore */
export * from './TeamUserSkeleton';
