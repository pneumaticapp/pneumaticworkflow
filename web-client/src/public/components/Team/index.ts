export * from './Team';
// eslint-disable-next-line no-restricted-exports
export { TeamContainer as default } from './container';
