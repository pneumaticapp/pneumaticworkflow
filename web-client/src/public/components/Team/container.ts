import { connect } from 'react-redux';
import { loadMicrosoftInvites } from "../../redux/team/actions";

import { IApplicationState } from '../../types/redux';
import {
  loadChangeUserAdmin,
  teamFetchStarted,
  resetUsers,
  changeUserListSorting,
  openDeleteUserModal,
  openTeamInvitesPopup,
  setGeneralLoaderVisibility,
} from '../../redux/actions';
import { ITeamProps, Team } from './Team';
import { withSyncedQueryString } from '../../HOCs/withSyncedQueryString';
import { EUserListSorting } from '../../types/user';
import { getIsUserSubsribed } from '../../redux/selectors/user';

export type TTeamProps = Pick<
  ITeamProps,
  | 'currentUserId'
  | 'isLoading'
  | 'sorting'
  | 'users'
  | 'isTeamInvitesOpened'
  | 'isSubscribed'
  | 'trialEnded'
  | 'userListSorting'
>;
type TTeamDispatchProps = Pick<
  ITeamProps,
  | 'fetchUsers'
  | 'resetUsers'
  | 'loadChangeUserAdmin'
  | 'openModal'
  | 'openTeamInvitesPopup'
  | 'setGeneralLoaderVisibility'
  | 'loadMicrosoftInvites'
>;

export function mapStateToProps(state: IApplicationState): TTeamProps {
  const {
    accounts,
    authUser,
    accounts: {
      planInfo: { trialEnded },
    },
    teamInvites: { isInvitesPopupOpen: isTeamInvitesOpened },
  } = state;
  const isSubscribed = getIsUserSubsribed(state);

  return {
    users: accounts.team.list,
    currentUserId: Number(authUser.id),
    isLoading: accounts.team.isLoading,
    sorting: accounts.userListSorting,
    isTeamInvitesOpened,
    isSubscribed,
    userListSorting: accounts.userListSorting,
    trialEnded,
  };
}

export const mapDispatchToProps: TTeamDispatchProps = {
  fetchUsers: teamFetchStarted,
  loadChangeUserAdmin,
  resetUsers,
  openModal: openDeleteUserModal,
  openTeamInvitesPopup,
  setGeneralLoaderVisibility,
  loadMicrosoftInvites,
};

const SyncedTeam = withSyncedQueryString<ITeamProps>([
  {
    propName: 'userListSorting',
    queryParamName: 'sorting',
    defaultAction: changeUserListSorting(EUserListSorting.NameAsc),
    createAction: changeUserListSorting,
    getQueryParamByProp: (value) => value,
  },
])(Team);

export const TeamContainer = connect<TTeamProps, TTeamDispatchProps>(mapStateToProps, mapDispatchToProps)(SyncedTeam);
