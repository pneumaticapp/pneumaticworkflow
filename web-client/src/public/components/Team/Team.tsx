import * as React from 'react';
import { useIntl } from 'react-intl';

import { TeamUserSkeleton } from './TeamUserSkeleton';
import { TeamUser } from './TeamUser';
import { DeleteTeamUserPopupContainer } from './DeleteTeamUserPopup';
import { AddGuestsBanner } from './AddGuestsBanner';

import { resendInvite } from '../../api/resendInvite';
import { TITLES } from '../../constants/titles';
import { EUserListSorting, TUserListItem } from '../../types/user';
import { getErrorMessage } from '../../utils/getErrorMessage';
import {
  IChangeUserAdminProps,
  TOpenDeleteUserModalPayload,
  ITeamFetchStartedProps,
} from '../../redux/accounts/actions';
import { IntlMessages } from '../IntlMessages';
import { NotificationManager } from '../UI/Notifications';
import { trackInviteTeamInPage } from '../../utils/analytics';
import { RoundPlusIcon, SearchLargeIcon } from '../icons';
import { Header, InputField } from '../UI';

import styles from './Team.css';

export const STATUS_TITLE_MAP: { [key: string]: string } = {
  active: 'team.title-group-active',
  invited: 'team.title-group-invited',
};

export function Team({
  currentUserId,
  isLoading,
  users,
  isSubscribed,
  fetchUsers,
  resetUsers,
  loadChangeUserAdmin,
  openModal,
  openTeamInvitesPopup,
  setGeneralLoaderVisibility,
  loadMicrosoftInvites
}: ITeamProps) {
  const { formatMessage } = useIntl();

  const [searchQuery, setSearchQuery] = React.useState('');

  const renderSkeleton = () => Array.from([1, 2, 3], (key) => <TeamUserSkeleton key={key} />);

  React.useEffect(() => {
    document.title = TITLES.Team;

    fetchUsers({});
    loadMicrosoftInvites();

    return () => {
      resetUsers();
    };
  }, []);

  const handleClickInviteButton = () => {
    openTeamInvitesPopup();
    trackInviteTeamInPage('Team Page');
  };

  const renderInviteButton = () => {
    return (
      <button type="button" onClick={handleClickInviteButton} className={styles['invite-button']}>
        <RoundPlusIcon className={styles['invite-button__icon']} />
        <IntlMessages id="team.invite-team-large-btn">
          {(text) => (
            <Header tag="span" size="6">
              {text}
            </Header>
          )}
        </IntlMessages>
      </button>
    );
  };

  const renderUsers = () => {
    if (isLoading) {
      return renderSkeleton();
    }

    return getFilteredUsers().map((user) => {
      return (
        <TeamUser
          key={user.id}
          resendInvite={resendInviteHandler(user.id)}
          isCurrentUser={currentUserId === user.id}
          isSubscribed={isSubscribed}
          handleToggleAdmin={handleToggleUserAdmin}
          openModal={() => openModal({ user })}
          user={{ ...user, isAccountOwner: user.isAccountOwner && !user.invite }}
        />
      );
    });
  };

  const resendInviteHandler = (userId?: number) => async () => {
    if (userId) {
      try {
        setGeneralLoaderVisibility(true);
        await resendInvite(userId);
        NotificationManager.success({ message: 'team.resend-invite-success-msg' });
      } catch (err) {
        NotificationManager.warning({ message: getErrorMessage(err) });
      } finally {
        setGeneralLoaderVisibility(false);
      }
    }
  };

  const handleToggleUserAdmin = (user: TUserListItem) => async () => {
    const { id, isAdmin, email } = user;

    if (id && email) {
      loadChangeUserAdmin({ id, isAdmin: !isAdmin, email });
    }
  };

  const getFilteredUsers = () => {
    if (!searchQuery) {
      return users;
    }

    const queryWords = searchQuery.split(' ').map((str) => str.toLowerCase());
    const checkQueryWord = (queryWord: string, user: TUserListItem) => {
      return [user.firstName, user.lastName, user.email]
        .map((str) => str.toLowerCase())
        .some((property) => property.includes(queryWord));
    };

    const filteredUsers = users.filter((user) => {
      return queryWords.every((queryWord) => checkQueryWord(queryWord, user));
    });

    return filteredUsers;
  };

  return (
    <div className={styles['container']}>
      <DeleteTeamUserPopupContainer />
      <AddGuestsBanner />

      <div className={styles['search']}>
        <SearchLargeIcon className={styles['search__icon']} />
        <InputField
          value={searchQuery}
          onChange={(e) => setSearchQuery(e.currentTarget.value)}
          containerClassName={styles['search-field']}
          className={styles['search-field__input']}
          placeholder={formatMessage({ id: 'team.search' })}
          fieldSize="md"
          onClear={() => setSearchQuery('')}
        />
      </div>

      {renderInviteButton()}

      <div className={styles['cards']}>{renderUsers()}</div>
    </div>
  );
}

export interface ITeamProps {
  currentUserId: number;
  isLoading?: boolean;
  sorting?: EUserListSorting;
  users: TUserListItem[];
  isTeamInvitesOpened?: boolean;
  isSubscribed?: boolean;
  trialEnded: boolean | null;
  userListSorting: EUserListSorting;
  fetchUsers(payload: ITeamFetchStartedProps): void;
  resetUsers(): void;
  loadChangeUserAdmin(payload: IChangeUserAdminProps): void;
  openModal(params: TOpenDeleteUserModalPayload): void;
  openTeamInvitesPopup(): void;
  loadMicrosoftInvites(): void;
  setGeneralLoaderVisibility(isVisible: boolean): void;
}
