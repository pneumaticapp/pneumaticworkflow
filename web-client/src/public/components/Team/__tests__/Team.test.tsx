import * as React from 'react';
import { shallow } from 'enzyme';

import { Team, ITeamProps } from '../Team';
import { mockLastNameUsers } from '../../../__stubs__/users';
import { EUserListSorting } from '../../../types/user';
import { TeamUser } from '../TeamUser';

const mockProps: ITeamProps = {
  currentUserId: 0,
  fetchUsers: jest.fn(),
  resetUsers: jest.fn(),
  loadChangeUserAdmin: jest.fn(),
  openModal: jest.fn(),
  users: mockLastNameUsers,
  userListSorting: EUserListSorting.NameAsc,
  openTeamInvitesPopup: jest.fn(),
  setGeneralLoaderVisibility: jest.fn(),
  trialEnded: null,
  loadMicrosoftInvites: jest.fn(),
};

describe('Team', () => {
  it('rendered correctly', () => {
    const wrapper = shallow(<Team {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('rendered correctly with descending sorting', () => {
    const wrapper = shallow(<Team {...mockProps} sorting={EUserListSorting.NameDesc} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('correcltly reacts on user id', () => {
    const wrapper = shallow(<Team {...mockProps} />);

    expect(wrapper.find(TeamUser).at(0).props().isCurrentUser).toBe(true);
    expect(wrapper.find(TeamUser).at(2).props().isCurrentUser).toBe(false);
  });
});
