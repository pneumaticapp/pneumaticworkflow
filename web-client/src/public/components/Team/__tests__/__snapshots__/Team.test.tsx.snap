// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team rendered correctly 1`] = `
<div
  className="container"
>
  <[object Object] />
  <AddGuestsBanner />
  <div
    className="search"
  >
    <SearchLargeIcon
      className="search__icon"
    />
    <InputField
      className="search-field__input"
      containerClassName="search-field"
      fieldSize="md"
      onChange={[Function]}
      onClear={[Function]}
      placeholder="Search user"
      value=""
    />
  </div>
  <button
    className="invite-button"
    onClick={[Function]}
    type="button"
  >
    <RoundPlusIcon
      className="invite-button__icon"
    />
    <ForwardRef
      id="team.invite-team-large-btn"
    >
      <Component />
    </ForwardRef>
  </button>
  <div
    className="cards"
  >
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={true}
      key="0"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_0.app",
          "firstName": "Stas",
          "id": 0,
          "isAccountOwner": undefined,
          "lastName": undefined,
          "phone": "88005553535",
          "photo": "/img/avatar0.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="1"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_1.app",
          "firstName": "Stas",
          "id": 1,
          "isAccountOwner": undefined,
          "lastName": "Boreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar1.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="2"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_2.app",
          "firstName": "Stas",
          "id": 2,
          "isAccountOwner": undefined,
          "lastName": "Careckiy",
          "phone": "88005553535",
          "photo": "/img/avatar2.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="3"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_3.app",
          "firstName": "Stas",
          "id": 3,
          "isAccountOwner": undefined,
          "lastName": "Tvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar3.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="4"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_4.app",
          "firstName": "Stas",
          "id": 4,
          "isAccountOwner": undefined,
          "lastName": "Skvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar4.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="5"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_5.app",
          "firstName": "Stas",
          "id": 5,
          "isAccountOwner": undefined,
          "lastName": "",
          "phone": "88005553535",
          "photo": "/img/avatar5.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="6"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_6.app",
          "firstName": "Stas",
          "id": 6,
          "isAccountOwner": undefined,
          "lastName": "Boreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar6.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="7"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_7.app",
          "firstName": "Stas",
          "id": 7,
          "isAccountOwner": undefined,
          "lastName": "Careckiy",
          "phone": "88005553535",
          "photo": "/img/avatar7.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="8"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_8.app",
          "firstName": "Stas",
          "id": 8,
          "isAccountOwner": undefined,
          "lastName": "Tvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar8.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="9"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_9.app",
          "firstName": "Stas",
          "id": 9,
          "isAccountOwner": undefined,
          "lastName": "Skvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar9.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="10"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_10.app",
          "firstName": "Stas",
          "id": 10,
          "isAccountOwner": undefined,
          "lastName": "",
          "phone": "88005553535",
          "photo": "/img/avatar10.png",
          "status": "active",
          "type": "user",
        }
      }
    />
  </div>
</div>
`;

exports[`Team rendered correctly with descending sorting 1`] = `
<div
  className="container"
>
  <[object Object] />
  <AddGuestsBanner />
  <div
    className="search"
  >
    <SearchLargeIcon
      className="search__icon"
    />
    <InputField
      className="search-field__input"
      containerClassName="search-field"
      fieldSize="md"
      onChange={[Function]}
      onClear={[Function]}
      placeholder="Search user"
      value=""
    />
  </div>
  <button
    className="invite-button"
    onClick={[Function]}
    type="button"
  >
    <RoundPlusIcon
      className="invite-button__icon"
    />
    <ForwardRef
      id="team.invite-team-large-btn"
    >
      <Component />
    </ForwardRef>
  </button>
  <div
    className="cards"
  >
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={true}
      key="0"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_0.app",
          "firstName": "Stas",
          "id": 0,
          "isAccountOwner": undefined,
          "lastName": undefined,
          "phone": "88005553535",
          "photo": "/img/avatar0.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="1"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_1.app",
          "firstName": "Stas",
          "id": 1,
          "isAccountOwner": undefined,
          "lastName": "Boreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar1.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="2"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_2.app",
          "firstName": "Stas",
          "id": 2,
          "isAccountOwner": undefined,
          "lastName": "Careckiy",
          "phone": "88005553535",
          "photo": "/img/avatar2.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="3"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_3.app",
          "firstName": "Stas",
          "id": 3,
          "isAccountOwner": undefined,
          "lastName": "Tvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar3.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="4"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_4.app",
          "firstName": "Stas",
          "id": 4,
          "isAccountOwner": undefined,
          "lastName": "Skvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar4.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="5"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_5.app",
          "firstName": "Stas",
          "id": 5,
          "isAccountOwner": undefined,
          "lastName": "",
          "phone": "88005553535",
          "photo": "/img/avatar5.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="6"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_6.app",
          "firstName": "Stas",
          "id": 6,
          "isAccountOwner": undefined,
          "lastName": "Boreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar6.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="7"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_7.app",
          "firstName": "Stas",
          "id": 7,
          "isAccountOwner": undefined,
          "lastName": "Careckiy",
          "phone": "88005553535",
          "photo": "/img/avatar7.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="8"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_8.app",
          "firstName": "Stas",
          "id": 8,
          "isAccountOwner": undefined,
          "lastName": "Tvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar8.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="9"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_9.app",
          "firstName": "Stas",
          "id": 9,
          "isAccountOwner": undefined,
          "lastName": "Skvoreckiy",
          "phone": "88005553535",
          "photo": "/img/avatar9.png",
          "status": "active",
          "type": "user",
        }
      }
    />
    <TeamUser
      handleToggleAdmin={[Function]}
      isCurrentUser={false}
      key="10"
      openModal={[Function]}
      resendInvite={[Function]}
      user={
        {
          "email": "stas@boreckiy_10.app",
          "firstName": "Stas",
          "id": 10,
          "isAccountOwner": undefined,
          "lastName": "",
          "phone": "88005553535",
          "photo": "/img/avatar10.png",
          "status": "active",
          "type": "user",
        }
      }
    />
  </div>
</div>
`;
